parallel Load/index create: open file exists, SIGSEGV

<!--

Welcome! - We kindly ask that you:

  1. Fill out the issue template below - not doing so needs a good reason.
  2. Use the forum if you have a question rather than a bug or feature request.

The forum is at: https://forum.restic.net

NOTE: Not filling out the issue template needs a good reason, as otherwise it
may take a lot longer to find the problem, not to mention it can take up a lot
more time which can otherwise be spent on development. Please also take the
time to help us debug the issue by collecting relevant information, even if
it doesn't seem to be relevant to you. Thanks!

The forum is a better place for questions about restic or general suggestions
and topics, e.g. usage or documentation questions! This issue tracker is mainly
for tracking bugs and feature requests directly relating to the development of
the software itself, rather than the project.

Thanks for understanding, and for contributing to the project!

-->


Output of `restic version`
--------------------------
`restic 0.9.5 compiled with go1.11.6 on linux/amd64`

```
cat /etc/os-release
NAME=NixOS
ID=nixos
VERSION="19.03.173147.77295b0bd26 (Koi)"
VERSION_CODENAME=koi
VERSION_ID="19.03.173147.77295b0bd26"
PRETTY_NAME="NixOS 19.03.173147.77295b0bd26 (Koi)"
LOGO="nix-snowflake"
HOME_URL="https://nixos.org/"
SUPPORT_URL="https://nixos.org/nixos/support.html"
BUG_REPORT_URL="https://github.com/NixOS/nixpkgs/issues"
```

How did you run restic exactly?
-------------------------------
```
$ export RESTIC_REPOSITORY=/mnt/restic-old
$ export RESTIC_PASSWORD_FILE=/home/edwin/restic-old
$ cat all.list | grep 2-home|sudo -E xargs -n 1 -P 8 ./bup-to-restic.sh
```
Here is `bup-to-restic.sh`:
```
#!/bin/sh
set -eux
BUP=/tmp/enc/bup2
DIR="/tmp/fast/$1/"
mkdir -p "$DIR"

bup -d "$BUP" restore -C "$DIR" --numeric-ids "$1"
restic backup --one-file-system --tag old --tag bup "$DIR"
rm "$DIR" -rf
```

```
+ bup -d /tmp/enc/bup2 restore -C /tmp/fast//c14-sd-92016-2-home/2017-06-16-233342/ --numeric-ids /c14-sd-92016-2-home/2017-06-16-233342
Restoring: 1065563, done.
Restoring: 1066171, done.
Restoring: 1066311, done.
Restoring: 1065399, done.
Restoring: 1065785, done.
Restoring: 1065973, done.
+ restic backup --one-file-system --tag old --tag bup /tmp/fast//c14-sd-92016-2-home/2017-06-15-233952/
+ restic backup --one-file-system --tag old --tag bup /tmp/fast//c14-sd-92016-2-home/2017-06-16-233342/
+ restic backup --one-file-system --tag old --tag bup /tmp/fast//c14-sd-92016-2-home/2017-06-12-233359/
+ restic backup --one-file-system --tag old --tag bup /tmp/fast//c14-sd-92016-2-home/2017-06-11-233357/
+ restic backup --one-file-system --tag old --tag bup /tmp/fast//c14-sd-92016-2-home/2017-06-14-233416/
+ restic backup --one-file-system --tag old --tag bup /tmp/fast//c14-sd-92016-2-home/2017-06-13-233400/
repository 4e6f982c opened successfully, password is correct
created new cache in /home/edwin/.cache/restic
repository 4e6f982c opened successfully, password is correct
repository 4e6f982c opened successfully, password is correct
repository 4e6f982c opened successfully, password is correct
repository 4e6f982c opened successfully, password is correct
repository 4e6f982c opened successfully, password is correct
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 552.330144ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 552.330144ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 552.330144ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 552.330144ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 552.330144ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 720.254544ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 720.254544ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 720.254544ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 720.254544ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 582.280027ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 582.280027ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 582.280027ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 552.330144ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 720.254544ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 582.280027ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 720.254544ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 468.857094ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 468.857094ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 582.280027ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 468.857094ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 693.478123ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 693.478123ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 693.478123ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 890.117305ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 703.28564ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 890.117305ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 873.42004ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 890.117305ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 703.28564ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 424.227764ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 693.478123ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 424.227764ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 693.478123ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 424.227764ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 492.389441ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 492.389441ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 890.117305ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 492.389441ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 890.117305ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 703.28564ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 671.590708ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 671.590708ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 671.590708ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 901.025843ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 901.025843ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 901.025843ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 1.142114207s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 1.142114207s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 1.142114207s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 636.341646ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 636.341646ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 738.584161ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 738.584161ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 1.040217184s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 1.335175957s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 1.477844956s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 1.477844956s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 1.477844956s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 671.590708ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 671.590708ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 1.205320285s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 1.205320285s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 1.205320285s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 1.351538765s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 1.351538765s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 1.713171311s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 1.713171311s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 1.486109007s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 1.486109007s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 1.486109007s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 2.216767434s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 2.216767434s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 1.380473169s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 1.380473169s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 1.380473169s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 954.512469ms: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 1.107876242s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 2.45250242s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 2.45250242s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 2.45250242s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 1.32137013s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 1.32137013s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 1.32137013s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 1.807980427s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 1.807980427s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 1.511079094s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 2.007539076s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 2.007539076s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 2.007539076s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 2.22916351s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 2.22916351s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 2.984558086s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 2.984558086s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 2.984558086s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 2.027308147s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 1.818837414s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 1.818837414s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 1.818837414s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 2.070709754s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 2.070709754s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x77fd18]

goroutine 27 [running]:
github.com/restic/restic/internal/repository.(*Index).ID(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        internal/repository/index.go:377 +0x38
github.com/restic/restic/internal/repository.(*Repository).LoadIndex.func5(0x8, 0xe0a7a0)
        internal/repository/repository.go:467 +0xd4
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc0003a0600, 0xc00000b1e0)
        vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x57
created by golang.org/x/sync/errgroup.(*Group).Go
        vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x66
+ BUP=/tmp/enc/bup2
+ DIR=/tmp/fast//c14-sd-92016-2-home/2017-06-17-233341/
+ mkdir -p /tmp/fast//c14-sd-92016-2-home/2017-06-17-233341/
+ bup -d /tmp/enc/bup2 restore -C /tmp/fast//c14-sd-92016-2-home/2017-06-17-233341/ --numeric-ids /c14-sd-92016-2-home/2017-06-17-233341
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 3.67875363s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 3.67875363s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 2.669912672s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 2.669912672s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 2.669912672s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 3.268621161s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 3.268621161s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 3.268621161s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 2.973082793s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 2.973082793s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 4.065212595s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 4.065212595s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 4.065212595s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 3.011308614s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 3.011308614s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 5.173209677s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 5.173209677s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 5.173209677s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 4.517033471s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 4.517033471s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 4.517033471s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 6.715255694s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 6.715255694s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 4.539634288s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 4.539634288s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 4.539634288s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 4.092384182s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 4.092384182s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 4.004869008s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 4.004869008s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 7.133794867s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 7.133794867s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 7.133794867s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 4.024209066s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 4.024209066s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 4.024209066s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 11.66401435s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 11.66401435s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 11.66401435s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 7.354397612s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 7.354397612s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 10.223534436s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 10.223534436s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 10.223534436s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 9.146728339s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 9.146728339s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 11.639721773s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 11.639721773s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 8.74646488s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 8.74646488s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 8.74646488s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 4.513276731s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 4.513276731s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 4.513276731s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x77fd18]

goroutine 81 [running]:
github.com/restic/restic/internal/repository.(*Index).ID(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        internal/repository/index.go:377 +0x38
github.com/restic/restic/internal/repository.(*Repository).LoadIndex.func5(0x8, 0xe0a7a0)
        internal/repository/repository.go:467 +0xd4
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc0003a83c0, 0xc00033f960)
        vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x57
created by golang.org/x/sync/errgroup.(*Group).Go
        vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x66
+ BUP=/tmp/enc/bup2
+ DIR=/tmp/fast//c14-sd-92016-2-home/2017-06-18-233345/
+ mkdir -p /tmp/fast//c14-sd-92016-2-home/2017-06-18-233345/
+ bup -d /tmp/enc/bup2 restore -C /tmp/fast//c14-sd-92016-2-home/2017-06-18-233345/ --numeric-ids /c14-sd-92016-2-home/2017-06-18-233345
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 10.163325309s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 8.436116856s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 8.436116856s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 8.436116856s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 10.214177147s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 14.188847299s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 14.188847299s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 14.188847299s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 18.904414573s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 18.904414573s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 18.904414573s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 7.425381294s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 7.425381294s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 7.425381294s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 16.05103845s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 21.044060183s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 21.044060183s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/0114bbc039>, 0, 0) returned error, retrying after 21.044060183s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/0114bbc03978b2bae119bba39df7897e9863788e089b73c4518b9f1910e2892d: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 13.581705599s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 10.747238066s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 10.747238066s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/01b5f7f8b4>, 0, 0) returned error, retrying after 10.747238066s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/01/01b5f7f8b4311c8af23691a101f11f4c709a208dd89d43a0be802b3dfd975f70: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 26.244032286s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 22.912714797s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 22.912714797s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 22.912714797s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05cb62375b>, 0, 0) returned error, retrying after 15.406612236s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05cb62375b9a63dd480f4c03856770b9e897b6b210221f6e155de9150313053b: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 15.406612236s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
Load(<index/05f7c2ddaf>, 0, 0) returned error, retrying after 15.406612236s: Create: open /home/edwin/.cache/restic/4e6f982c6f79ad117b00689be08ae28f092514b274830525af025a231a0951c5/index/05/05f7c2ddafbede6d5df3a8f7518398f24fd6a4e79419d8dba7a701c4f0b83169: file exists
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x77fd18]

goroutine 29 [running]:
github.com/restic/restic/internal/repository.(*Index).ID(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        internal/repository/index.go:377 +0x38
github.com/restic/restic/internal/repository.(*Repository).LoadIndex.func5(0xc000000008, 0xe0a7a0)
        internal/repository/repository.go:467 +0xd4
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc000275d10, 0xc0003a3060)
        vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x57
created by golang.org/x/sync/errgroup.(*Group).Go
        vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x66
+ BUP=/tmp/enc/bup2
+ DIR=/tmp/fast//c14-sd-92016-2-home/2017-06-19-233343/
+ mkdir -p /tmp/fast//c14-sd-92016-2-home/2017-06-19-233343/
+ bup -d /tmp/enc/bup2 restore -C /tmp/fast//c14-sd-92016-2-home/2017-06-19-233343/ --numeric-ids /c14-sd-92016-2-home/2017-06-19-233343

Files:       1064954 new,     0 changed,     0 unmodified
Dirs:            3 new,     0 changed,     0 unmodified
Added to the repo: 388.533 MiB

processed 1064954 files, 20.123 GiB in 7:50
snapshot b5616219 saved
+ rm /tmp/fast//c14-sd-92016-2-home/2017-06-16-233342/ -rfes 20.114 GiB, 0 errors ETA 0:28

Files:       1064814 new,     0 changed,     0 unmodified
Dirs:            3 new,     0 changed,     0 unmodified
Added to the repo: 387.280 MiB

processed 1064814 files, 20.120 GiB in 7:58
snapshot 62036c75 saved

Files:       1064616 new,     0 changed,     0 unmodified
Dirs:            3 new,     0 changed,     0 unmodified
Added to the repo: 387.240 MiB

processed 1064616 files, 20.114 GiB in 7:59
snapshot 16e4ff62 saved
```
<!--
This section should include at least:

 * The complete command line and any environment variables you used to
   configure restic's backend access. Make sure to replace sensitive values!

 * The output of the commands, what restic prints gives may give us much
   information to diagnose the problem!
-->

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
Local disk, ext4 HDD:
```
mount|grep /mnt
/dev/sdb1 on /mnt type ext4 (rw,noatime,nodiratime)
```

The temporary extraction space is ext4 too (nvme SSD).

Expected behavior
-----------------

Not crash

<!--
Describe what you'd like restic to do differently.
-->

Actual behavior
---------------

SIGSEGV, first time I see restic crash and I backed up ~1.4TiB with it already.
However it crashed repeatedly now in quick succession.
<!--
In this section, please try to concentrate on observations, so only describe
what you observed directly.
-->

Steps to reproduce the behavior
-------------------------------

I didn't try to reproduce the behaviour, but it reproduced itself at least once, see above.

<!--
The more time you spend describing an easy way to reproduce the behavior (if
this is possible), the easier it is for the project developers to fix it!
-->

Do you have any idea what may have caused this?
-----------------------------------------------
I am running restic in parallel, could that be a problem? Certainly EEXIST should be expected from the cache, unless restic uses locking.


Do you have an idea how to solve the issue?
-------------------------------------------

Ignore EEXIST for cache files, instead of retrying to create fthe file which is now always going to fail with EEXIST retry at a higher level. Sorry I can't offer more details than this, I'm not familiar with the code.

Did restic help you or made you happy in any way?
-------------------------------------------------

Seems to be faster than be able to cope with large backups better than `bup`, and `borg`, hence I'm migrating my backups to `restic`. `bup` becomes very slow on large backups, takes ages to even just list what you have in the backup, and borg was a bit memory heavy and not so good at paralleizing.
The borg migration went pretty smooth, the bup one is the first one where restic had som e hiccups.

<!--
Answering this question is not required, but if you have anything positive to share, please do so here!
Sometimes we get tired of reading bug reports all day and a little positive end note does wonders.
Idea by Joey Hess, https://joeyh.name/blog/entry/two_holiday_stories/
-->

--- a/changelog/unreleased/issue-2345
+++ b/changelog/unreleased/issue-2345
@@ -1,0 +1,12 @@
+Bugfix: Make cache crash-resistant and usable by multiple concurrent processes
+
+The restic cache directory ($RESTIC_CACHE_DIR) could end up in a broken state
+in the event of restic (or the OS) crashing. This is now less likely to occur
+as files are downloaded to a temporary location before being moved in place.
+
+This also allows multiple concurrent restic processes to operate on a single
+repository without conflicts. Previously, concurrent operations could cause
+segfaults because the processes saw each other's partially downloaded files.
+
+https://github.com/restic/restic/issues/2345
+https://github.com/restic/restic/pull/2838

--- a/internal/cache/backend.go
+++ b/internal/cache/backend.go
@@ -43,14 +43,13 @@ func (b *Backend) Remove(ctx context.Context, h restic.Handle) error {
 	return b.Cache.remove(h)
 }
 
-var autoCacheTypes = map[restic.FileType]struct{}{
-	restic.IndexFile:    {},
-	restic.SnapshotFile: {},
+func autoCached(t restic.FileType) bool {
+	return t == restic.IndexFile || t == restic.SnapshotFile
 }
 
 // Save stores a new file in the backend and the cache.
 func (b *Backend) Save(ctx context.Context, h restic.Handle, rd restic.RewindReader) error {
-	if _, ok := autoCacheTypes[h.Type]; !ok {
+	if !autoCached(h.Type) {
 		return b.Backend.Save(ctx, h, rd)
 	}
 
@@ -82,11 +81,6 @@ func (b *Backend) Save(ctx context.Context, h restic.Handle, rd restic.RewindRea
 	}
 
 	return nil
-}
-
-var autoCacheFiles = map[restic.FileType]bool{
-	restic.IndexFile:    true,
-	restic.SnapshotFile: true,
 }
 
 func (b *Backend) cacheFile(ctx context.Context, h restic.Handle) error {
@@ -192,7 +186,7 @@ func (b *Backend) Load(ctx context.Context, h restic.Handle, length int, offset 
 	}
 
 	// if we don't automatically cache this file type, fall back to the backend
-	if _, ok := autoCacheFiles[h.Type]; !ok {
+	if !autoCached(h.Type) {
 		debug.Log("Load(%v, %v, %v): delegating to backend", h, length, offset)
 		return b.Backend.Load(ctx, h, length, offset, consumer)
 	}

--- a/internal/cache/cache.go
+++ b/internal/cache/cache.go
@@ -28,7 +28,7 @@ const fileMode = 0644
 
 func readVersion(dir string) (v uint, err error) {
 	buf, err := ioutil.ReadFile(filepath.Join(dir, "version"))
-	if os.IsNotExist(err) {
+	if errors.Is(err, os.ErrNotExist) {
 		return 0, nil
 	}
 
@@ -61,13 +61,13 @@ func writeCachedirTag(dir string) error {
 
 	tagfile := filepath.Join(dir, "CACHEDIR.TAG")
 	_, err := fs.Lstat(tagfile)
-	if err != nil && !os.IsNotExist(err) {
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
 		return errors.WithStack(err)
 	}
 
 	f, err := fs.OpenFile(tagfile, os.O_CREATE|os.O_EXCL|os.O_WRONLY, fileMode)
 	if err != nil {
-		if os.IsExist(errors.Cause(err)) {
+		if errors.Is(err, os.ErrExist) {
 			return nil
 		}
 
@@ -121,7 +121,7 @@ func New(id string, basedir string) (c *Cache, err error) {
 	// create the repo cache dir if it does not exist yet
 	var created bool
 	_, err = fs.Lstat(cachedir)
-	if os.IsNotExist(err) {
+	if errors.Is(err, os.ErrNotExist) {
 		err = fs.MkdirAll(cachedir, dirMode)
 		if err != nil {
 			return nil, errors.WithStack(err)
@@ -179,11 +179,10 @@ func validCacheDirName(s string) bool {
 // listCacheDirs returns the list of cache directories.
 func listCacheDirs(basedir string) ([]os.FileInfo, error) {
 	f, err := fs.Open(basedir)
-	if err != nil && os.IsNotExist(errors.Cause(err)) {
-		return nil, nil
-	}
-
 	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			err = nil
+		}
 		return nil, err
 	}
 

--- a/internal/cache/file.go
+++ b/internal/cache/file.go
@@ -99,7 +99,7 @@ func (c *Cache) Save(h restic.Handle, rd io.Reader) error {
 	finalname := c.filename(h)
 	dir := filepath.Dir(finalname)
 	err := fs.Mkdir(dir, 0700)
-	if err != nil && !os.IsExist(err) {
+	if err != nil && !errors.Is(err, os.ErrExist) {
 		return err
 	}
 

--- a/internal/cache/file.go
+++ b/internal/cache/file.go
@@ -2,8 +2,10 @@ package cache
 
 import (
 	"io"
+	"io/ioutil"
 	"os"
 	"path/filepath"
+	"runtime"
 
 	"github.com/pkg/errors"
 	"github.com/restic/restic/internal/crypto"
@@ -84,31 +86,26 @@ func (c *Cache) load(h restic.Handle, length int, offset int64) (io.ReadCloser, 
 	return rd, nil
 }
 
-// SaveWriter returns a writer for the cache object h. It must be closed after writing is finished.
-func (c *Cache) saveWriter(h restic.Handle) (io.WriteCloser, error) {
-	debug.Log("Save to cache: %v", h)
-	if !c.canBeCached(h.Type) {
-		return nil, errors.New("cannot be cached")
-	}
-
-	p := c.filename(h)
-	err := fs.MkdirAll(filepath.Dir(p), 0700)
-	if err != nil {
-		return nil, errors.WithStack(err)
-	}
-
-	f, err := fs.OpenFile(p, os.O_CREATE|os.O_EXCL|os.O_WRONLY, 0400)
-	return f, errors.WithStack(err)
-}
-
 // Save saves a file in the cache.
 func (c *Cache) Save(h restic.Handle, rd io.Reader) error {
 	debug.Log("Save to cache: %v", h)
 	if rd == nil {
 		return errors.New("Save() called with nil reader")
 	}
+	if !c.canBeCached(h.Type) {
+		return errors.New("cannot be cached")
+	}
 
-	f, err := c.saveWriter(h)
+	finalname := c.filename(h)
+	dir := filepath.Dir(finalname)
+	err := fs.Mkdir(dir, 0700)
+	if err != nil && !os.IsExist(err) {
+		return err
+	}
+
+	// First save to a temporary location. This allows multiple concurrent
+	// restics to use a single cache dir.
+	f, err := ioutil.TempFile(dir, "tmp-")
 	if err != nil {
 		return err
 	}
@@ -116,23 +113,38 @@ func (c *Cache) Save(h restic.Handle, rd io.Reader) error {
 	n, err := io.Copy(f, rd)
 	if err != nil {
 		_ = f.Close()
-		_ = c.remove(h)
+		_ = fs.Remove(f.Name())
 		return errors.Wrap(err, "Copy")
 	}
 
 	if n <= crypto.Extension {
 		_ = f.Close()
-		_ = c.remove(h)
+		_ = fs.Remove(f.Name())
 		debug.Log("trying to cache truncated file %v, removing", h)
 		return nil
 	}
 
+	// Close, then rename. Windows doesn't like the reverse order.
 	if err = f.Close(); err != nil {
-		_ = c.remove(h)
+		_ = fs.Remove(f.Name())
 		return errors.WithStack(err)
 	}
 
-	return nil
+	err = fs.Rename(f.Name(), finalname)
+	if err != nil {
+		_ = fs.Remove(f.Name())
+	}
+	if runtime.GOOS == "windows" && errors.Is(err, os.ErrPermission) {
+		// On Windows, renaming over an existing file is ok
+		// (os.Rename is MoveFileExW with MOVEFILE_REPLACE_EXISTING
+		// since Go 1.5), but not when someone else has the file open.
+		//
+		// When we get Access denied, we assume that's the case
+		// and the other process has written the desired contents to f.
+		err = nil
+	}
+
+	return errors.WithStack(err)
 }
 
 // Remove deletes a file. When the file is not cache, no error is returned.

--- a/internal/cache/file_test.go
+++ b/internal/cache/file_test.go
@@ -3,14 +3,17 @@ package cache
 import (
 	"bytes"
 	"fmt"
-	"io"
 	"io/ioutil"
 	"math/rand"
+	"os"
 	"testing"
 	"time"
 
+	"github.com/restic/restic/internal/errors"
 	"github.com/restic/restic/internal/restic"
 	"github.com/restic/restic/internal/test"
+
+	"golang.org/x/sync/errgroup"
 )
 
 func generateRandomFiles(t testing.TB, tpe restic.FileType, c *Cache) restic.IDSet {
@@ -131,64 +134,6 @@ func TestFiles(t *testing.T) {
 	}
 }
 
-func TestFileSaveWriter(t *testing.T) {
-	seed := time.Now().Unix()
-	t.Logf("seed is %v", seed)
-	rand.Seed(seed)
-
-	c, cleanup := TestNewCache(t)
-	defer cleanup()
-
-	// save about 5 MiB of data in the cache
-	data := test.Random(rand.Int(), 5234142)
-	id := restic.ID{}
-	copy(id[:], data)
-	h := restic.Handle{
-		Type: restic.PackFile,
-		Name: id.String(),
-	}
-
-	wr, err := c.saveWriter(h)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	n, err := io.Copy(wr, bytes.NewReader(data))
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if n != int64(len(data)) {
-		t.Fatalf("wrong number of bytes written, want %v, got %v", len(data), n)
-	}
-
-	if err = wr.Close(); err != nil {
-		t.Fatal(err)
-	}
-
-	rd, err := c.load(h, 0, 0)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	buf, err := ioutil.ReadAll(rd)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if len(buf) != len(data) {
-		t.Fatalf("wrong number of bytes read, want %v, got %v", len(data), len(buf))
-	}
-
-	if !bytes.Equal(buf, data) {
-		t.Fatalf("wrong data returned, want:\n  %02x\ngot:\n  %02x", data[:16], buf[:16])
-	}
-
-	if err = rd.Close(); err != nil {
-		t.Fatal(err)
-	}
-}
-
 func TestFileLoad(t *testing.T) {
 	seed := time.Now().Unix()
 	t.Logf("seed is %v", seed)
@@ -256,4 +201,56 @@ func TestFileLoad(t *testing.T) {
 			}
 		})
 	}
+}
+
+// Simulate multiple processes writing to a cache, using goroutines.
+func TestFileSaveConcurrent(t *testing.T) {
+	const nproc = 40
+
+	c, cleanup := TestNewCache(t)
+	defer cleanup()
+
+	var (
+		data = test.Random(1, 10000)
+		g    errgroup.Group
+		id   restic.ID
+	)
+	rand.Read(id[:])
+
+	h := restic.Handle{
+		Type: restic.PackFile,
+		Name: id.String(),
+	}
+
+	for i := 0; i < nproc/2; i++ {
+		g.Go(func() error { return c.Save(h, bytes.NewReader(data)) })
+
+		// Can't use load because only the main goroutine may call t.Fatal.
+		g.Go(func() error {
+			// The timing is hard to get right, but the main thing we want to
+			// ensure is ENOENT or nil error.
+			time.Sleep(time.Duration(100+rand.Intn(200)) * time.Millisecond)
+
+			f, err := c.load(h, 0, 0)
+			t.Logf("Load error: %v", err)
+			switch {
+			case err == nil:
+			case errors.Is(err, os.ErrNotExist):
+				return nil
+			default:
+				return err
+			}
+			defer func() { _ = f.Close() }()
+
+			read, err := ioutil.ReadAll(f)
+			if err == nil && !bytes.Equal(read, data) {
+				err = errors.New("mismatch between Save and Load")
+			}
+			return err
+		})
+	}
+
+	test.OK(t, g.Wait())
+	saved := load(t, c, h)
+	test.Equals(t, data, saved)
 }

--- a/internal/fs/file.go
+++ b/internal/fs/file.go
@@ -40,6 +40,14 @@ func RemoveAll(path string) error {
 	return os.RemoveAll(fixpath(path))
 }
 
+// Rename renames (moves) oldpath to newpath.
+// If newpath already exists, Rename replaces it.
+// OS-specific restrictions may apply when oldpath and newpath are in different directories.
+// If there is an error, it will be of type *LinkError.
+func Rename(oldpath, newpath string) error {
+	return os.Rename(fixpath(oldpath), fixpath(newpath))
+}
+
 // Symlink creates newname as a symbolic link to oldname.
 // If there is an error, it will be of type *LinkError.
 func Symlink(oldname, newname string) error {

